app/screens/ScheduleScreen/ScheduleScreen.tsx

```diff
@@ -131,6 +165,48 @@ export const ScheduleScreen: React.FC<TabScreenProps<"Schedule">> = () => {
     [isFocused],
   )
 
+  const currentEventIndex = useSharedValue(-1)
+  const currentlyViewingEvents = useSharedValue<number[]>([])
+  const currentlyViewingSchedule = useSharedValue(0)
+
+  const handleViewableEventIndexChanged = useRef(
+    ({ viewableItems }: { viewableItems: RNViewToken[] }) => {
+      currentlyViewingEvents.value = viewableItems.map((item) => item.index)
+    },
+  ).current
+
+  const handleViewableScheduleIndexChanged = useRef(
+    ({ viewableItems }: { viewableItems: RNViewToken[] }) => {
+      currentlyViewingSchedule.value = viewableItems[0]?.index ?? 0
+    },
+  ).current
+
+  const scrollButtonOpacity = useDerivedValue(() => {
+    return withSpring(
+      scheduleIndex === currentlyViewingSchedule.value &&
+        currentEventIndex.value > -1 &&
+        currentlyViewingEvents.value[0] !== currentEventIndex.value
+        ? 1
+        : 0,
+    )
+  }, [scheduleIndex])
+
+  const $scrollButtonStyle = useAnimatedStyle(() => ({
+    opacity: scrollButtonOpacity.value,
+  }))
+
+  const $arrowStyle = useAnimatedStyle(() => ({
```

Comment:
You don't need to do anything for this PR, but at some point, I want to see if we can refactor and component-ize this schedule screen a bit so there's less going on in this one file. I think with the right props, we could extract this new button at least, and possibly the list itself.

Reply:
Agreed!